odoo/addons/base/models/ir_http.py

```diff
@@ -186,6 +186,25 @@ def _pre_dispatch(cls, rule, args):
         env = request.env if request.env.uid else request.env['base'].with_user(SUPERUSER_ID).env
         request.update_context(lang=get_lang(env)._get_cached('code'))
 
+        for key, val in list(args.items()):
+            if not isinstance(val, models.BaseModel):
+                continue
+
+            try:
+                # explicitly crash now, instead of crashing later
+                args[key].check_access_rights('read')
+                args[key].check_access_rule('read')
+            except (odoo.exceptions.AccessError, odoo.exceptions.MissingError) as e:
+                if isinstance(e, odoo.exceptions.MissingError):
+                    new_e = werkzeug.exceptions.NotFound()
+                    new_e.__cause__ = e
+                    e = new_e
+                # custom behavior in case a record is not accessible / has been removed
+                if handle_error := rule.endpoint.original_routing.get('handle_params_access_error'):
+                    if response := handle_error(e):
+                        werkzeug.exceptions.abort(response)
+                raise e from None
+
```

Comment:
Sorry to keep asking you changing things, it now comes to my attention that we are going to `handle_error(NotFound())` in case of a `MissingError`, API-wise I think it makes more sense to propagate the MissingError to handle_error.

Inverting the two also allow for a simplification regarding exception reraising

```suggestion
            except (odoo.exceptions.AccessError, odoo.exceptions.MissingError) as e:
                # custom behavior in case a record is not accessible / has been removed
                if handle_error := rule.endpoint.original_routing.get('handle_params_access_error'):
                    if response := handle_error(e):
                        werkzeug.exceptions.abort(response)
                if isinstance(e, odoo.exceptions.MissingError):
                    raise werkzeug.exceptions.NotFound() from e
                raise

```

Reply:
done